### PR TITLE
Enable/disable cling ptrcheck through annotations

### PR DESCRIPTION
**Pull request by Axel-Naumann**

### CODE REVIEW

**Comment by oshadura** [comment]
@phsft-bot build!

**Comment by bluehood** [comment]
more benchmarks are coming!

**Comment by bluehood** [comment]
TL;DR: all benchmarks benefit from the patch, no other change in behavior observed

Intel(R) Core(TM) i7-4790 CPU @ 3.60GHz (4 physical cores)
All times in seconds.

### `TTree::AsMatrix` benchmark (see [ROOT-9377](https://sft.its.cern.ch/jira/browse/ROOT-9377))
#### Master (no patch)
```                                                                                                               
total interpreted: 5.23 
loop interpreted:  4.1 
compile -O3:       3.2 
execute loop:      0.14
````
These timings are consistent with the latest that were performed on this benchmark.                                                                                             
             
#### With patch
```                                                                                                         
total interpreted: 1.5
loop interpreted:  0.37 
compilation and loop execution: same as above (master no patch)
```
Interpreted loop takes the same time as with a non-optimized g++ compilation.

### Fons' benchmark (PROOF vs RDF, see [ROOT-9538](https://sft.its.cern.ch/jira/browse/ROOT-9538))
#### Master (no patch)
```
---- run_dataframe.C ----
interpreted loop: 8
compile O3: 5
compiled loop: 5

--- run_dataframe_imt.C (4 threads) ---
interpreted loop: 6
compile O3: 5
compiled loop: 3
```
#### With patch
```
--- run_dataframe.C ---
interpreted: 5
compile -O3: 5
loop: 5

--- run_dataframe_imt.C (4 threads) ---
interpreted: 3
compile -O3: 5
loop: 3
```

**Comment by dpiparo** [comment]
`df102_NanoAODDimuonAnalysis.C` - before 3m30s -> after 1m08s
